jerry-core/ecma/operations/ecma-array-object.c

```diff
@@ -173,7 +173,7 @@ ecma_op_new_array_object_from_length (ecma_length_t length) /**< length of the n
 #if ENABLED (JERRY_ESNEXT)
   if (length > UINT32_MAX)
   {
-    ecma_raise_range_error (ECMA_ERR_MSG ("Invalid array length"));
+    ecma_raise_range_error (ECMA_ERR_MSG ("Invalid Array length"));
```

Comment:
Object is often rewritten to object, but array is almost always rewritten to Array? Why this duality?

Reply:
There should be some logic behind naming. My current preference is that primitive types (can be the result of a `typeof` operator) are lowercase, and specialized objects are uppercase. So `number` (value) and `Number` (object) are two different things. The exception is `BigInt`, which looks ugly in lowercase and `Symbol` because of v8. I don't like the latter, but I have no choice. There is no `array` in ECMAScript only a specialized `Array` object.